src/components/Modals/CreateSingleItemModal/CreateSingleItemModal.tsx

```diff
@@ -286,10 +288,16 @@ export default class CreateSingleItemModal extends React.PureComponent<Props, St
           metrics,
           contents: await computeHashes(sortedContents.all),
           createdAt: +new Date(),
-          updatedAt: +new Date()
+          updatedAt: +new Date(),
+          serverContentHash: null
         }
       }
 
+      // Add the catalyst image to the item
+      const catalystImage = await generateCatalystImage(item, { thumbnail: sortedContents.all[THUMBNAIL_PATH] })
+      sortedContents.all[IMAGE_PATH] = catalystImage.content
+      item.contents[IMAGE_PATH] = catalystImage.hash
```

Comment:
Does it make sense to abstract these 3 lines ? they seem similar to these https://github.com/decentraland/builder/pull/1820/files#diff-58a4f4bd05bdec4de69216a6c54b669acd456a2f36a93f05d47a5315f53febb9R141 but I'm unsure if it's accidental duplication

Reply:
They seem similar, but they're generating an image and setting its content and hash to different objects. It might not add much to abstract that logic.